### PR TITLE
Upgrade checkstyle

### PR DESCRIPTION
**Pull request by valfirst**
## Change list

 - Upgrade to Checkstyle 8.7
 - Fix checkstyle violations found after version update
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [X] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/appium/java_client/pagefactory/AppiumFieldDecorator.java [hunk 1]
```diff
@@ -79,11 +78,11 @@ public AppiumFieldDecorator(SearchContext context, long timeout,
     }
 
     /**
-     * @param context is an instance of {@link org.openqa.selenium.SearchContext}
-     *                It may be the instance of {@link org.openqa.selenium.WebDriver}
-     *                or {@link org.openqa.selenium.WebElement} or
-     *                {@link io.appium.java_client.pagefactory.Widget} or some other user's
-     *                extension/implementation.
+     * Creates field decorator based on serach {@code context} and timeout {@code duration}.
```
Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/pagefactory/AppiumFieldDecorator.java]: should be `SearchContext`

File: google-style.xml [hunk 2]
```diff
@@ -46,19 +46,16 @@
...
+        </module>
         <module name="OneTopLevelClass"/>
         <module name="NoLineWrap"/>
         <module name="EmptyBlock">
             <property name="option" value="TEXT"/>
             <property name="tokens" value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
         </module>
         <module name="NeedBraces"/>
-        <module name="LeftCurly">
-            <property name="maxLineLength" value="100"/>
```
Comment (SrinivasanTarget) [inline google-style.xml]: Was it intentionally removed?

File: build.gradle [hunk 3]
```diff
@@ -128,19 +128,12 @@ tasks.withType(JacocoReport) {
 jacocoTestReport.dependsOn test
 
 checkstyle {
-    toolVersion = '7.0'
+    toolVersion = '8.7'
+    configFile = file("$projectDir/google-style.xml")
     showViolations = true
```
Comment (SrinivasanTarget) [inline build.gradle]: 👍

File: google-style.xml (hunk 2 above)
Comment (valfirst) [reply google-style.xml]: yes, it was deprecated for a long time and finally it was removed in [Checkstyle 8.2](http://checkstyle.sourceforge.net/releasenotes.html#Release_8.2): https://github.com/checkstyle/checkstyle/issues/3671

**Comment by TikhomirovSergey** [comment]
@SrinivasanTarget I will check it today.

**Pull request merged**